### ISSUE TITLE
Admin settings Key metrics answer preview loading issue.

### ISSUE THREAD

**Issue by mohitwp** [google/site-kit-wp#6428]
## Bug Description

If analytics is connected then for milliseconds "CTA gets load" before we see the answer preview screen.

## Steps to reproduce

1. Provide answers to User Input Questions.
2. Set up Analytics.
3. Go to Admin settings.
4. Notice before Question and Answers preview 'Key metrics CTA' gets load for milliseconds.

## Screenshots

https://user-images.githubusercontent.com/94359491/212305531-0eec3d22-f951-4a3a-adc9-7b7c15ef5357.mp4

## Additional Context

 - PHP Version: 
 - Browser: [e.g. chrome, safari]
 - Plugin Version: [e.g. 22]

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* There should be no flicker that shows User Input Admin CTA in the Key Metrics section of Settings page. 
    * Note that is is very hard to spot, throttle connection speed to observe it. 

## Implementation Brief

* The flickers appears to be happening due to `isUserInputCompleted` briefly returning `undefined` before resolving while the other prerequisites are resolved and met. (ie. Both Analytics and SC is connected and has data). 
* To account for this, return null from the `UserInputSettings` component when `isUserInputCompleted` is `undefined` or `'true'` 

### Test Coverage

* 

## QA Brief

* Provide answers to User Input Questions.
* Set up Analytics.
* Go to Admin settings.
* Flicker should be eliminated 

## Changelog entry

* Fix key metrics preview loading issue on the admin settings page.

**Comment by eugene-manuilov** [comment]
IB ✔️

**Comment by aaemnnosttv** [comment]
@kuasha420 @eugene-manuilov `userInputState` was recently removed / repurposed into `isUserInputCompleted` so this my either no longer be an issue or at least need to be updated.

**Comment by kuasha420** [comment]
@aaemnnosttv Can you link me to the issue where this happened? I assume it hasn't been merged into develop yet?

**Comment by kuasha420** [comment]
Nevermind, found the issue #5900 and PR #6119. I'll wait for #6119 to be merged before having another look at this, I'll keep this one with me until then. Cheers

**Comment by kuasha420** [comment]
@aaemnnosttv Checked and confirmed the issue still exists with the latest `develop` and updated the IB accordingly. Moving straight to EB.

**Comment by mohitwp** [comment]
### QA Update ❌ 

@sashadoes 

- Tested on dev environment. 

- When user navigates to admin settings, then I can notice flicker just before key metrics answer preview. Now CTA is not showing, but it loads plugin status before then display key metrics settings. You can clearly see this in below video. Pause video at 0.04 seconds and forward it manually.


https://user-images.githubusercontent.com/94359491/216073808-5c5c35fc-41ef-43c0-b60a-6f214898267e.mp4

**Comment by wpdarren** [comment]
@mohitwp good spot. I spotted the same when I was testing the key metrics banner on the main dashboard. I was informed by Jimmy that this would be fixed in another ticket.  You can [read about that here](https://github.com/google/site-kit-wp/issues/5900#issuecomment-1411874094). It looks like 6209 and 6210 would fix this, but I will let @sashadoes confirm that for admin settings.

**Comment by sashadoes** [comment]
Hi @mohitwp. After the check, I can confirm that the implementation I made is not connected to the above issue in Account Settings. So should be fixed as @wpdarren stated.

**Comment by wpdarren** [comment]
@jimmymadon Hi, the issue I found with the flicker of the key metrics banner on the dashboard is also appearing on the admin settings screen.  Will the tickets you [mentioned here](https://github.com/google/site-kit-wp/issues/5900#issuecomment-1411874094) also fix this?  I don't want us to approve this and then find out it'll not be fixed.  Thank you!   c.c. @mohitwp

**Comment by mohitwp** [comment]
### QA Update ✅ 

-Tested on dev environment.
- Issue reported[ above](https://github.com/google/site-kit-wp/issues/6428#issuecomment-1412178729) is now fixed.
- Key metrics answer preview now loading fine under admin settings.

<details>

https://user-images.githubusercontent.com/94359491/217143161-a4ecb5f8-12f9-4335-98ce-f948f8e2a04b.mp4

</details>

**Comment by mohitwp** [comment]
@sashadoes It would be good if we can add progress bar also when user refreshes the page. Not only when user switch to admin settings tab. @jimmymadon @wpdarren Can you please suggest what ideal behavior we can achieve in this case ?

**Comment by wpdarren** [comment]
@sashadoes I agree with @mohitwp it looks a odd UX/UI when the progress bar loads for a second then the answers to the questions load when you refresh the page. Especially because it doesn't occur when you click on the admin settings. Both should behave the same in my opinion.  Interested in @jimmymadon opinion on this.  It's minor, but will likely come up in the bug bash so better to resolve it now.

**Comment by aaemnnosttv** [comment]
After discussing with @jimmymadon, we agreed to quickly fix the second case he mentioned above related to gathering state conditions.

We'll open a new issue about addressing flickers due to "super quick" progress/loaders more holistically 👍

**Comment by aaemnnosttv** [comment]
After some further investigation with @jimmymadon, we found that the conditional logic to display the user input CTA on the admin settings based on module gathering states was a bit premature and will inevitably lead to a flicker or shift any time we need to check if the modules are in a gathering state that requires an API request (regardless if we check for gathering or not gathering). We've gone ahead and removed that conditional logic for now, with the intention to reintroduce it in (or after) #5933 which ideally should have been done first as this will provide the gathering state on page load so long as we know that the module is _not_ in a gathering state. So we'll follow up to reintroduce this after that issue.

Another reason for deferring this is related to a few existing tests which were passing because of the flicker (CTA seen as present) before disappearing because of the lack of gathering state being considered/provided in the test. This change brings the interface back into parity with the expectations of the test without requiring additional time-consuming changes to fix.

TL;DR there is no more flicker on the admin settings because the user input CTA should be shown regardless of the gathering state now, but this will be corrected in a future issue.

**Comment by aaemnnosttv** [comment]
@mohitwp please see the updates outlined above, otherwise this should be ready for another pass 👍

**Comment by mohitwp** [comment]
### QA Update ✅ 


- Tested on main.
- Verified that if site is in gathering state and analytics is connected, then CTA is showing under Admin settings page and on SK dashboard.
- Verified that no flicker showing related to CTA under Admin settings.

<details>

https://user-images.githubusercontent.com/94359491/217739272-afd4e490-1750-4600-8ac7-97bcaccafa35.mp4


https://user-images.githubusercontent.com/94359491/217739377-274ce9b9-5b84-4b4c-a3d0-892f49e7082a.mp4

</details>
